aperturedb/ParallelQuery.py

```diff
@@ -103,7 +159,30 @@ def worker(self, thid, generator, start, end):
                 self.pb.update((i + 1) / total_batches)
 
     def query(self, generator, batchsize=1, numthreads=4, stats=False):
-
+        """
+        This function takes as input the data to be executed in specified number of threads.
+        The generator yields a tuple : (array of commands, array of blobs)
+
+        Args:
+            generator (_type_): _description_
+            batchsize (int, optional): _description_. Defaults to 1.
+            numthreads (int, optional): _description_. Defaults to 4.
+            stats (bool, optional): _description_. Defaults to False.
+        """
+        if len(generator) > 0:
+            if isinstance(generator[0], tuple) and isinstance(generator[0][0], list):
```

Comment:
mmm why is any of this needed? Wouldn't the generators always return [q, [blobs]] ? I will ask you when we chat.

Reply:
The old data loaders do not follow that. Tests will fail.